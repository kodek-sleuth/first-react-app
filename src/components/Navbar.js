import React from 'react';
import { Link } from 'react-router-dom';
import Proptypes from 'prop-types';

const Navbar = ({ title, icon }) => {
    return (
        <div className="navbar">
            <h1>{ title }<img src="https://image.flaticon.com/icons/svg/270/270798.svg" alt="" className="githubIcon" /> </h1>
            <div>
                <ul>
                    <li>
                        <Link to='/'>Home</Link>
                    </li>
                    <li>
                        <Link to='/search'>Search</Link>
                    </li>
                </ul> 
            </div>
        </div>
    )
}

Navbar.defaultProps = {
    title: 'Github Finder',
    icon: 'fab fa-github'
}

Navbar.propTypes = {
    title: Proptypes.string.isRequired,
    icon: Proptypes.string.isRequired
}

export default Navbar
