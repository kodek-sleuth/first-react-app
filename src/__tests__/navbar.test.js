import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Navbar from '../components/Navbar';

describe('<Navbar />', () => {
    it('should render the navbar', () => {
        const navbar = create(
        <MemoryRouter>
            <Navbar />
        </MemoryRouter>)
        expect(navbar.toJSON()).toMatchSnapshot();
    });
});

