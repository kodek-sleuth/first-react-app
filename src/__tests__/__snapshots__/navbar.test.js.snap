// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render the navbar 1`] = `
<div
  className="navbar"
>
  <h1>
    Github Finder
    <img
      alt=""
      className="githubIcon"
      src="https://image.flaticon.com/icons/svg/270/270798.svg"
    />
     
  </h1>
  <div>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </li>
    </ul>
  </div>
</div>
`;
